### PR TITLE
Fixing #5932 TouchableOpacity respect style opacity

### PR DESCRIPTION
**Pull request by gorangajic**
Initial `opacity` on `TouchableOpacity` Component is broken like it's pointed in issue #5932

### CODE REVIEW

**Comment by facebook-github-bot-3** [comment]
By analyzing the blame information on this pull request, we identified @janicduplessis and @jesseruder to be potential reviewers.

**Comment by gorangajic** [comment]
I have update pull request because there is also a bug when `opacity` is updated later it will remain the same.

**Comment by gorangajic** [comment]
Anything I can do to get this merged faster?

**Comment by r1cebank** [comment]
I would love to see this merged soon.

**Comment by facebook-github-bot-4** [comment]
It's been a while since the last commit was reviewed and the labels show this pull request needs review. Based on the blame information for the files in this pull request we identified @janicduplessis as a potential reviewer. Could you take a look please or cc someone with more context?

**Comment by gustavjf** [comment]
I've just added this to Product Pains on RN support page. Same title as #5932

**Comment by facebook-github-bot-0** [comment]
@gorangajic updated the pull request - [view changes](https://github.com/facebook/react-native/pull/8909/files/0effe5ef05234c9b69334f10497e5e1a4a4ab489..90761ee86fb368a73939d4be4c7bcc4c68d30b99)

File: Libraries/Components/Touchable/TouchableOpacity.js [hunk 1]
```diff
@@ -98,6 +106,7 @@ var TouchableOpacity = React.createClass({
   touchableHandleActivePressIn: function(e: Event) {
     this.clearTimeout(this._hideTimeout);
     this._hideTimeout = null;
+    this._pressIn = true;
```
Comment (mkonicek) [inline Libraries/Components/Touchable/TouchableOpacity.js]: Does this variable exist on TouchableOpacity or are you introducing a new one?

File: Libraries/Components/Touchable/TouchableOpacity.js [hunk 2]
```diff
@@ -67,16 +67,24 @@ var TouchableOpacity = React.createClass({
...
 
   componentDidMount: function() {
     ensurePositiveDelayProps(this.props);
   },
 
+  componentDidUpdate: function() {
+    if (!this._hideTimeout && !this._pressIn) {
+      this._opacityInactive();
+    }
+  },
```
Comment (mkonicek) [inline Libraries/Components/Touchable/TouchableOpacity.js]: Why is this needed?

File: Libraries/Components/Touchable/TouchableOpacity.js [hunk 3]
```diff
@@ -67,16 +67,24 @@ var TouchableOpacity = React.createClass({
   },
 
   getInitialState: function() {
+    var childStyle = flattenStyle(this.props.style) || {};
+    var opacity = childStyle.opacity === undefined ? 1 : childStyle.opacity;
     return {
       ...this.touchableGetInitialState(),
-      anim: new Animated.Value(1),
+      anim: new Animated.Value(opacity),
```
Comment (mkonicek) [inline Libraries/Components/Touchable/TouchableOpacity.js]: Makes sense, thanks!

File: Libraries/Components/Touchable/TouchableOpacity.js (hunk 1 above)
Comment (gorangajic) [inline Libraries/Components/Touchable/TouchableOpacity.js]: introducing a new one

File: Libraries/Components/Touchable/TouchableOpacity.js (hunk 2 above)
Comment (gorangajic) [inline Libraries/Components/Touchable/TouchableOpacity.js]: This is needed because one edge case that I stumble upon. When you are in the "pressing" state if component update it will change opacity, so to prevent that we need to ignore that update and update only when press is released

not sure if I am clear, if you need more info I will try to explain that better

**Comment by RyanMitchellWilson** [comment]
Is this still being tracked? Would really like this in the next update if possible.

**Comment by T-Spoon** [comment]
Just ran into this bug - definitely would be nice to get this one in, considering the footprint of this PR is quite small.

**Comment by hramos** [comment]
@mkonicek are you satisfied with the latest changes from the author?

File: Libraries/Components/Touchable/TouchableOpacity.js (hunk 2 above)
Comment (lacker) [inline Libraries/Components/Touchable/TouchableOpacity.js]: Can you add a test to ensure that this continues to work after you get your fix in?

**Comment by lacker** [comment]
I think this sort of bug fix is a good time to add a test, because it seems quite easy for someone else to inadvertently break this bugfix later.

**Comment by evollu** [comment]
can anyone review this? @janicduplessis and @jesseruder?

**Comment by Kielan** [comment]
@janicduplessis @mkonicek @jesseruder pinging you guys again for the PR, please merge or give more feedback so we can know what needs to further be done.

**Comment by hramos** [comment]
@evollu @Kielan see above feedback from @lacker - this PR should add a test before it can be merged in.

**Comment by ryankask** [comment]
It looks like #12628 fixes the initial opacity but doesn't address the edge case mentioned above. Should this PR be closed/updated to address that edge case?

**Comment by ryankask** [comment]
It looks like the "edge case" mentioned above is important because if the opacity value changes, it should be updated.

**Comment by hellogerard** [comment]
> It looks like the "edge case" mentioned above is important because if the opacity value changes, it should be updated.
 
Agree. I have run into this as well (and it took some sleuthing to find this thread!). This PR is superior to #12628, IMO.
